lib/circlex/api/payments/bank_accounts.ex

```diff
@@ -54,6 +56,47 @@ defmodule Circlex.Api.Payments.BankAccounts do
     end
   end
 
+  @doc """
+  Create a bank account (wires) for an international bank account that supports list_bank_accounts
+
+    iex> host = Circlex.Test.start_server()
+    iex> billing_details = %{city: "Toronto", country: "CA", district: "ON", line1: "100 Money St", name: "Satoshi Nakamoto", postalCode: "ON M5J 1S9"}
+    iex> bank_address = %{bankName: "HSBC Canada", city: "Toronto", country: "CA"}
+    iex> Circlex.Api.Payments.BankAccounts.create_non_us_iban_supported("1000000001", billing_details, bank_address, host: host)
+    {
+      :ok,
+      %Circlex.Struct.BankAccount{
+        description: "HSBC Canada 0001",
+        bank_address: %{"bankName" => "HSBC Canada", "city" => "Toronto", "country" => "CA"},
+        billing_details: %{"city" => "Toronto", "country" => "CA", "district" => "ON", "line1" => "100 Money St", "name" => "Satoshi Nakamoto", "postalCode" => "ON M5J 1S9"},
+        create_date: "2022-07-17T08:59:41.344582Z",
+        fingerprint: "b09eb536-05ae-11ed-aaa8-6a1733211c01",
+        id: "a033a6d8-05ae-11ed-9e62-6a1733211c00",
+        status: "pending",
+        tracking_ref: "CIR3KXZZ00",
+        update_date: "2022-07-17T08:59:41.344582Z"
+      }
+    }
+  """
+  @spec create_non_us_iban_supported(String.t(), map(), map(), list()) :: {:ok, %BankAccount{}}
```

Comment:
I love the typespec here, though we should have `| {:error, term()}`, right?

Reply:
Also, we should probably make `BankAccount.t()` at some point if we want.

Reply:
do you know if Thing.t is has a different behaviour than a struct here ? I get it when you're type is some composite type or you want to alias it or something, but %Struct{} is really easy to grok imo